llvm/lib/CodeGen/SelectionDAG/SelectionDAG.cpp

```diff
@@ -4017,6 +4015,16 @@ KnownBits SelectionDAG::computeKnownBits(SDValue Op, const APInt &DemandedElts,
       }
     }
 
+    // For SMIN, if CstHigh is negative we know the result will be
```

Comment:
Is this tested?

Reply:
Yes

Reply:
> Yes

Which test tests it? It wasn't obvious to me from the names.

Reply:
Don't "resolve conversations" initiated by someone else. Let them resolve it when they are satisifed.

Reply:
smin_known_zero_2

it mirrors smax's logic

Reply:
> smin_known_zero_2
> 
> it mirrors smax's logic

I don't see how that test could test something that uses ComputeNumSignBits.